src/routes/run_report.rs

```diff
@@ -226,17 +226,29 @@ async fn create(
     };
     // Get DB connection
     let conn = pool.get().expect("Failed to get DB connection from pool");
-    // Create run report
-    match report_builder
-        .create_run_report(
-            &conn,
-            id,
-            report_id,
-            &run_report_inputs.created_by,
-            delete_failed,
-        )
-        .await
-    {
+    // Get the report_builder and create the run report or return an error if we don't have one
+    match {
+        match report_builder.as_ref() {
+            Some(report_builder) => {
+                report_builder
+                    .create_run_report(
+                        &conn,
+                        id,
+                        report_id,
+                        &run_report_inputs.created_by,
+                        delete_failed,
+                    )
+                    .await
+            },
+            None => {
+                return Ok(HttpResponse::InternalServerError().json(ErrorBody{
+                    title: String::from("No report builder"),
+                    status: 500,
+                    detail: String::from("Reporting is configured but no report builder was constructed.  This is a bug.  Please complain about it on the carrot github.")
```

Comment:
Might as well include a link to our issues page here :P

Reply:
Oh good point